Sources/Contentful/ArrayResponse.swift

```diff
@@ -107,4 +111,193 @@ public struct MappedArrayResponse<ItemType>: Array where ItemType: EntryModellab
 
     /// The total number of resources which matched the original request
     public let total: UInt
+
+    internal let includes: MappedIncludes?
+
+    internal var includedAssets: [Asset]? {
+        return includes?.assets
+    }
+    internal var includedEntries: [EntryDecodable]? {
+        return includes?.entries
+    }
+}
+
+internal struct MappedIncludes: Decodable {
+    let assets: [Asset]?
+    let entries: [EntryDecodable]?
+
+    private enum CodingKeys: String, CodingKey {
+        case assets     = "Asset"
+        case entries    = "Entry"
+    }
+
+    init(from decoder: Decoder) throws {
+        let container   = try decoder.container(keyedBy: CodingKeys.self)
+
+        assets          = try container.decodeIfPresent([Asset].self, forKey: CodingKeys.assets)
+        // Cache to enable link resolution.
+        if let assets = assets {
+            decoder.linkResolver.cache(assets: assets)
+        }
+
+        // A copy as an array of dictionaries just to extract "sys.type" field.
+        guard let jsonItems = try container.decodeIfPresent(Swift.Array<Any>.self, forKey: .entries) as? [[String: Any]] else {
+            self.entries = nil
+            return
+        }
+        var entriesJSONContainer = try container.nestedUnkeyedContainer(forKey: .entries)
+        var entries: [EntryDecodable] = []
+        let contentTypes = decoder.userInfo[DecoderContext.contentTypesContextKey] as! [ContentTypeId: EntryDecodable.Type]
+
+        while entriesJSONContainer.isAtEnd == false {
```

Comment:
`while !entriesJSONContainer.isAtEnd {`

Reply:
I actually like `== false` here. It's easy to miss the ! at the start and it gets especially confusing when you have a semantic variable like `isAtEnd`. Just my 2c.

Reply:
yeah, i've chosen as the style for the sdk to use `== false` always over `!`

Reply:
I personally prefer conciseness, but I understand that it may get lost in the syntax.